### ISSUE TITLE
Coordinate masking to support stick reductions on padded tensors

### ISSUE THREAD

**Issue by avery-blanchard** [torch-spyre/torch-spyre#878]
### What happened?

Reducing a padded stick dimension requires coordinate masking. Currently a backend bug prevents this.

### What did you expect to happen?

Ability to reduce a stick dimension when padding is present.

### How can we reproduce it?

```
# min_failure.py
import torch
with torch.no_grad():

    x = torch.ones((3,3), dtype=torch.float16)
    DIM=(0,1)
    op = torch.amin

    compiled_fn = torch.compile(lambda a: op(a, dim=DIM), backend='sendnn')
```

### Any environmental details we need to know?

Standard environment

### Relevant log output

```Shell

```
